### PR TITLE
[DOC] Ruby command-line options

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Creates a new and comprehensive doc file for Ruby options;  removes two previous files that each covered some options.

I have positioned the file as `doc/ruby/options.md` b/c I want (later) to create a sibling file `doc/arguments.md`.

### CODE REVIEW

File: doc/ruby/options.md [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+"\r"
+$ ruby -0377 -e 'p $/'
+"\xFF"
+$ ruby -0400 -e 'p $/'
+nil
+```
+
+See also:
+
+- {Option -a}[options_md.html#label-a-3A+Split+Input+Lines+into+Fields]:
```
Comment (nobu) [inline doc/ruby/options.md:64]:
```suggestion
- {Option -a}[rdoc-ref:@a-3A+Split+Input+Lines+into+Fields]:
```

File: doc/ruby/options.md [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+
+See Thread::Backtrace.limit.
+
+### `--copyright`: Print Ruby Copyright
+
+Option `--copyright` prints a copyright message:
+
+```sh
+$ ruby --copyright
+ruby - Copyright (C) 1993-2021 Yukihiro Matsumoto
```
Comment (nobu) [inline doc/ruby/options.md:572]:
```suggestion
ruby - Copyright (C) 1993-2024 Yukihiro Matsumoto
```

File: doc/ruby/options.md (hunk 1 above)
Comment (BurdetteLamar) [reply doc/ruby/options.md:64]: This link did not work for me.  Try as I did, I could not find any rdoc-ref that would work with a heading that begins with a hyphen.

On my machiine:

```
$ rdoc --version
6.3.1
$ ruby --version
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]
```

**Comment by BurdetteLamar** [comment]
Should `options.md` be moved from `doc/ruby/` to `doc/ruby/command_line/`, to make room for more files?  I'm thinking in particular of:
```
- doc/
  - ruby/
    - command_line/
      - arguments.md    # Discuss what's an argument: in particular, -- and -s.
      - command_line.md # Discuss form; refer to options.md and arguments.md.
      - options.md      # The file offered in this PR.
    - program/          # User-written program. (This dir would need a better name.)
      - arguments.md:   # How to handle CL arguments (ARGV, ARGF, etc.).
      - options.md:     # How to handle CL options (OptionParser, GetoptLong, etc.).
```

File: doc/ruby/options.md (hunk 1 above)
Comment (nobu) [reply doc/ruby/options.md:64]: Use the latest and bundled version.

File: doc/ruby/options.md [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+the argument is a comma-separated list of the features to be disabled:
+
+```sh
+ruby --disable=gems,rubyopt t.rb
+```
+
+The supported features:
+
+- `gems`: Rubygems (default: enabled).
+- `did_you_mean`: `did_you_mean` (default: enabled).
```
Comment (peterzhu2118) [inline doc/ruby/options.md:592]: Maybe we can link to the gem:

```suggestion
- `did_you_mean`: [`did_you_mean`](https://github.com/ruby/did_you_mean) (default: enabled).
```

File: doc/ruby/options.md (hunk 1 above)
Comment (BurdetteLamar) [reply doc/ruby/options.md:64]: @nobu, what version of RDoc do you have where that link works?

Here's what I have (on WIndows), where it does not work.

```
$ ruby --version
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x64-mingw-ucrt]
$ rdoc --version
6.5.0
```

Comment (BurdetteLamar) [reply doc/ruby/options.md:64]: Also does not work with rdoc version 6.6.2 on Windows.  @peterzhu2118, does the link from @nobu work for you (and, regardless, what version are you running)?

Comment (peterzhu2118) [reply doc/ruby/options.md:64]: No, but this works for me:

```suggestion
- {Option -a}[rdoc-ref:ruby/options.md@a-3A+Split+Input+Lines+into+Fields]:
```

Comment (BurdetteLamar) [reply doc/ruby/options.md]: Fixed (all such).

**Comment by BurdetteLamar** [comment]
@peterzhu2118, is this ready to merge?

**Review by peterzhu2118 — APPROVED** [review]

**Comment by peterzhu2118** [comment]
@BurdetteLamar you need to rebase and address conflict to be able to merge.

**Comment by peterzhu2118** [comment]
You just need to address conflicts after rebasing. You can read about how to do that here: https://docs.github.com/en/get-started/using-git/resolving-merge-conflicts-after-a-git-rebase

**Comment by BurdetteLamar** [comment]
@peterzhu2118, this looks to me to be ready to merge ("no conflicts").  Is there a problem?

**Comment by peterzhu2118** [comment]
This is what I see.

Screenshot 2024-03-06 at 1 00 49 PM

**Comment by BurdetteLamar** [comment]
This is what I see:  This branch has no conflicts with the base branch.

**Comment by peterzhu2118** [comment]
I think you're rebasing on top of your master branch, which is older than Ruby's master branch. Try:

```
git pull --rebase https://github.com/ruby/ruby master
```

**Comment by peterzhu2118** [comment]
There should be conflicts you have to resolve.

**Comment by BurdetteLamar** [comment]
When I go to https://github.com/ruby/ruby/pull/10138 (and refresh), I see "This branch has no conflicts with the base branch."  What happens if I squash and merge?

**Comment by jeremyevans** [comment]
It looks like @BurdetteLamar merged instead of rebasing, since he pushed a merge commit.  Since there are no conflicts, even though this hasn't been rebased, squash and merge should work fine.  @peterzhu2118 would you be OK with squash and merge?

**Comment by nobu** [comment]
@BurdetteLamar There is still `environment.md` under `doc/command_line`, but it is excluded from document generation.
Is this intentional?

**Comment by BurdetteLamar** [comment]
> @BurdetteLamar There is still `environment.md` under `doc/command_line`, but it is excluded from document generation. Is this intentional?

Thanks, @nobu.

**Pull request merged**